homeassistant/components/deconz/cover.py

```diff
@@ -61,7 +61,7 @@ def __init__(self, device, gateway):
     @property
     def current_cover_position(self):
         """Return the current position of the cover."""
-        return 100 - int(self._device.brightness / 255 * 100)
+        return 100 - int(self._device.brightness / 254 * 100)
```

Comment:
Since `int()`truncates floating numbers towards zero the result would be the following.

Script:
```Python
### old behaviour

# reported max brightness 254, defined brightness 255
print(100 - int(254 / 255 * 100))

# reported max brightness 255, defined brightness 255
print(100 - int(255 / 255 * 100))

### new behaviour

# reported max brightness 254, defined brightness 254
print(100 - int(254 / 254 * 100))

# reported max brightness 255, defined brightness 254
print(100 - int(255 / 254 * 100))
```

Script output:
```
1
0
0
0
```

The expected result is `0`.

Reply:
btw: writing the test is probably beyond my non existing coding skills. Maybe I'll give it a try in the coming days.

Reply:
A test has been added.

Reply:
Great!